src/main/java/uk/gov/hmcts/reform/hmc/service/LinkedHearingGroupService.java

```diff
@@ -0,0 +1,124 @@
+        }
+    }
+
+    private void validateHearingGroupStatus(LinkedGroupDetails linkedGroupDetails) {
+        String groupStatus = linkedGroupDetails.getStatus();
+        if (invalidDeleteGroupStatuses.stream().anyMatch(e -> e.equals(groupStatus))) {
+            throw new BadRequestException(format(INVALID_DELETE_HEARING_GROUP_STATUS, groupStatus));
+        }
+    }
+
+    private void validateUnlinkingHearingsStatus(List<HearingEntity> linkedHearings) {
+        List<HearingEntity> unlinkInvalidStatusHearings = linkedHearings.stream()
+            .filter(h -> !DeleteHearingStatus.isValid(h.getStatus()))
+            .collect(Collectors.toList());
+
+        if (!unlinkInvalidStatusHearings.isEmpty()) {
+            throw new BadRequestException(
+                format(INVALID_DELETE_HEARING_GROUP_HEARING_STATUS, unlinkInvalidStatusHearings.get(0).getId()));
+        }
+    }
+
+    private void validateUnlinkingHearingsWillNotHaveStartDateInThePast(List<HearingEntity> linkedHearings) {
+
+        linkedHearings.stream()
+            .filter(h -> h.getHearingResponses().size() > 0)
+            .forEach(hearing -> {
+                List<HearingResponseEntity> latestVersionHearingResponses = getLatestVersionHearingResponses(hearing);
+
+                Optional<HearingResponseEntity> mostRecentLatestVersionHearingResponse = latestVersionHearingResponses
+                    .stream().max(Comparator.comparing(HearingResponseEntity::getRequestTimeStamp));
+
+                boolean hasHearingDateInThePast = mostRecentLatestVersionHearingResponse.isPresent()
+                    && mostRecentLatestVersionHearingResponse.get()
+                    .getHearingDayDetails().stream()
+                    .anyMatch(dayTime -> dayTime.getStartDateTime().isBefore(LocalDateTime.now()));
+
+                if (hasHearingDateInThePast) {
+                    throw new BadRequestException(format(
+                        INVALID_DELETE_HEARING_GROUP_DAY_IN_THE_PAST,
+                        hearing.getId()
+                    ));
+                }
+            });
+    }
+
+    private List<HearingResponseEntity> getLatestVersionHearingResponses(HearingEntity hearing) {
+        Optional<Map.Entry<String, List<HearingResponseEntity>>> max = hearing.getHearingResponses().stream()
+            .collect(groupingBy(HearingResponseEntity::getRequestVersion))
+            .entrySet()
+            .stream()
+            .max(Map.Entry.comparingByKey());
+
+        return max.isPresent() ? max.get().getValue() : List.of();
+    }
+
+    private void deleteFromLinkedGroupDetails(List<HearingEntity> linkedGroupHearings, Long hearingGroupId) {
+        linkedGroupHearings.forEach(hearingEntity -> {
+            // TODO: unlink hearingEntity from the group and persist - https://tools.hmcts.net/jira/browse/HMAN-96
+        });
+        linkedGroupDetailsRepository.deleteHearingGroup(hearingGroupId);
```

Comment:
as this method is not implemented in repository, how we are going to test this logic

Reply:
This has been added to the repository, so I could mock it in the service/ verify it has been invoked. And to test that if  the exception occurs it should not be invoked.